### PR TITLE
impl Arbitrary for IpAddr

### PR DESCRIPTION
**Pull request by jyn514**
This complements the existing impls for Ipv4Addr and Ipv6Addr.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -1140,6 +1140,20 @@ impl<'a> Arbitrary<'a> for Ipv6Addr {
     }
 }
 
+impl<'a> Arbitrary<'a> for IpAddr {
+    fn arbitrary(u: &mut Unstructured<'a>) -> Result<Self> {
+        if u.arbitrary()? {
+            u.arbitrary().map(IpAddr::V4)
+        } else {
+            u.arbitrary().map(IpAddr::V6)
+        }
```
Comment (fitzgen) [inline src/lib.rs:1149]: I find it much easier to read this code when written like this:

```suggestion
        if u.arbitrary()? {
            Ok(IpAddr::V4(u.arbitrary()?))
        } else {
            Ok(IpAddr::V6(u.arbitrary()?))
        }
```

File: src/lib.rs [hunk 2]
```diff
@@ -1140,6 +1140,20 @@ impl<'a> Arbitrary<'a> for Ipv6Addr {
...
+    }
+
+    fn size_hint(_depth: usize) -> (usize, Option<usize>) {
+        (4, Some(16))
```
Comment (fitzgen) [inline src/lib.rs:1153]: This version is robust to changes in how many bytes the sub-`Arbitrary` impls consume:

```suggestion
        size_hint::and(
            bool::size_hint(depth),
            size_hint::or(
                Ipv4Addr::size_hint(depth),
                Ipv6Addr::size_hint(depth),
            )
        )
```

**Review by fitzgen — COMMENTED** [review]
Thanks! Just a couple nitpicks to be addressed and we can merge this.

**Comment by jyn514** [comment]
thanks for the quick review! committed both suggestions :)

**Review by Manishearth — APPROVED** [review]

**Pull request merged**